sdk/js/src/token_bridge/transfer.ts

```diff
@@ -625,3 +630,152 @@ export async function transferFromAlgorand(
   txs.push({ tx: acTxn, signer: null });
   return txs;
 }
+
+/**
+ * Transfers an asset from Near to a receiver on another chain
+ * @param client
+ * @param tokenBridge account of the token bridge
+ * @param assetId account
+ * @param qty Quantity to transfer
+ * @param receiver Receiving account
+ * @param chain Reeiving chain
+ * @param fee Transfer fee
+ * @param payload payload for payload3 transfers
+ * @returns Sequence number of confirmation
+ */
+export async function transferTokenFromNear(
+  client: nearAccount,
+  coreBridge: string,
+  tokenBridge: string,
+  assetId: string,
+  qty: bigint,
+  receiver: Uint8Array,
+  chain: ChainId | ChainName,
+  fee: bigint,
+  payload: string = ""
+): Promise<[number, string]> {
+  let wormhole = assetId.endsWith("." + tokenBridge);
+
+  let result;
+
+  let message_fee = (await client.viewFunction(coreBridge, "message_fee", {}));
+
+  if (wormhole) {
+    result = await client.functionCall({
+      contractId: tokenBridge,
+      methodName: "send_transfer_wormhole_token",
+      args: {
+        token: assetId,
+        amount: qty.toString(10),
+        receiver: uint8ArrayToHex(receiver),
+        chain: chain,
+        fee: fee.toString(10),
+        payload: payload,
+        message_fee: message_fee,
+      },
+      attachedDeposit: new BN(message_fee + 1),
+      gas: new BN("100000000000000"),
+    });
+  } else {
+    let bal = await client.viewFunction(assetId, "storage_balance_of", {
+      account_id: tokenBridge,
+    });
+    if (bal == null) {
+      // Looks like we have to stake some storage for this asset
+      // for the token bridge...
+      nearAPI.providers.getTransactionLastResult(
+        await client.functionCall({
+          contractId: assetId,
+          methodName: "storage_deposit",
+          args: { account_id: tokenBridge, registration_only: true },
+          gas: new BN("100000000000000"),
+          attachedDeposit: new BN("2000000000000000000000"), // 0.002 NEAR
+        })
+      );
+    }
+
+    if (message_fee > 0) {
+      let bank = await client.viewFunction(tokenBridge, "bank_balance", { acct: client.accountId });
+
+      if (!bank[0]) {
+        await client.functionCall({
+          contractId: tokenBridge,
+          methodName: "register_bank",
+          args: {},
+          gas: new BN("100000000000000"),
+          attachedDeposit: new BN("2000000000000000000000"), // 0.002 NEAR
+        });
+      }
+
+      if (bank[1] < message_fee) {
+        await client.functionCall({
```

Comment:
Is there any way this can fail causing the subsequent transfer to fail?  Do we care?

Reply:
If we end up with a race condition, the ft_transfer_call will fail out and the transfer will "unwind" (hopefully).  You can then do the operation again without loss of funds